docs/data-model.rst

```diff
@@ -503,20 +503,48 @@ the library itself can use. All other information is considered to be
 **metadata**, and is stored in the ``metadata`` columns of the various
 tables.
 
-Arbitrary binary data can be stored in ``metadata`` columns, and the
-``tskit`` library makes no attempt to interpret this information. How the
-information held in this field is encoded is entirely the choice of client code.
+The metadata columns are :ref:`binary columns <sec_tables_api_binary_columns>`.
+However, each table with a ``metadata`` column also has a ``metadata_schema`` which
+describes the metadata and its encoding into binary. This schema is then used to
+validate, encode and decode the metadata. In this way the metadata is portable and
+self-describing. Methods such as :func:`TreeSequence.nodes`, ``node_table[56]`` and
+:func:`IndividualTable.add_row`  will use the schema to encode/decode the
+metadata for you. To use the bulk methods such as :func:`IndividualTable.set_columns`
+and :func:`IndividualTable.append_columns` you will need to encode the metadata as
+described :ref:`here <sec_tables_api_binary_columns_metadata>`
+
+Here is an example metadata schema that defines two properties, one of which is
+required and the other must match a regex. Unrecognised properties are not allowed,
+due to the ``additionalProperties`` flag. Note that descriptions can be added to help
+downstream users of your metadata.
+
+.. code-block:: python
+
+    import tskit
+
+    schema = tskit.metadata.JSONMetadataSchema(
+        {
+            "type": "object",
+            "properties": {
+                "accession_number": {"type": "number"},
+                "collection_date": {
+                    "name": "Collection date",
+                    "description": "Date of sample collection in ISO format",
+                    "type": "string",
+                    "pattern": "^([1-9][0-9]{3})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])?$",
+                },
+            },
+            "required": ["accession_number"],
+            "additionalProperties": False,
+        }
+    )
+
 
 To ensure that metadata can be safely interchanged using the :ref:`sec_text_file_format`,
-each row is `base 64 encoded <https://en.wikipedia.org/wiki/Base64>`_. Thus,
-binary information can be safely printed and exchanged, but may not be
+when using these each row is `base 64 encoded <https://en.wikipedia.org/wiki/Base64>`_.
```

Comment:
Not sure this sentence parses now

Reply:
Fixed https://github.com/tskit-dev/tskit/pull/543/commits/a6fcda86896547ea56b46e527a99d8401873c0aa